src/main/java/org/jabref/logic/util/io/FileNameUniqueness.java

```diff
@@ -41,4 +45,90 @@ public static String getNonOverWritingFileName(Path targetDirectory, String file
 
         return newFileName;
     }
+
+    /**
+     *
+     * @param directory The directory which saves the files (.pdf, for example)
+     * @param fileName Suggest name for the newly downloaded file
+     * @param dialogService To display the error and success message
+     * @return true when the content of files is duplicate and successfully delete the newly downloaded file ,
+     *         false when it is not a duplicate file or fail to delete the duplicate file
+     * @throws IOException Fail when the file is not exist or something wrong when reading the file
+     */
+    public static boolean isDuplicatedFile(Path directory, String fileName, DialogService dialogService) throws IOException {
+
+        Optional<String> extensionOptional = FileUtil.getFileExtension(fileName);
+        String extensionSuffix;
+        String fileNameWithoutDuplicated;
+
+        if (extensionOptional.isPresent()) {
+            extensionSuffix = '.' + extensionOptional.get();
+            fileNameWithoutDuplicated = eraseDuplicateMarks(fileName.substring(0, fileName.lastIndexOf('.')));
+        } else {
+            extensionSuffix = "";
+            fileNameWithoutDuplicated = eraseDuplicateMarks(fileName);
+        }
+
+        String originalFileName = fileNameWithoutDuplicated + extensionSuffix;
+        if (fileName.equals(originalFileName)) {
+            return false;
+        }
+
+        Path originalFile = Path.of(directory.toString(), originalFileName);
+        // deal with a very special case, when duplication does not happen, but the file name is end with something like " (1)" as it originally is
+        if (!Files.exists(directory.resolve(originalFileName))) {
+            return false;
+        }
+
+        Path duplicateFile = Path.of(directory.toString(), fileName);
+        int counter = 1;
+        while (true) {
+            if (com.google.common.io.Files.equal(originalFile.toFile(), duplicateFile.toFile())) {
+                if (duplicateFile.toFile().delete()) {
+                    dialogService.notify(Localization.lang("Dupilcate file with '%0', successfully delete the file '%1'", originalFileName, fileName));
+                } else {
+                    dialogService.notify(Localization.lang("Dupilcate file with '%0', fail to delete the file '%1'", originalFileName, fileName));
+                }
+                return true;
+            }
+
+            originalFileName = fileNameWithoutDuplicated +
+                    " (" + counter + ")"
+                    + extensionSuffix;
+            counter++;
+
+            if (originalFileName.equals(fileName)) {
+                dialogService.notify(Localization.lang("Duplicate file name but different content, keep the new file '%0'", fileName));
```

Comment:
I would remove this line, because the file name is not duplicate (but only simlar - hard to explain to the user)